workspaces/src/network/transaction.rs

```diff
@@ -0,0 +1,325 @@
+use std::convert::TryInto;
+
+use near_crypto::KeyType;
+use near_primitives::transaction::{
+    Action, AddKeyAction, CreateAccountAction, DeleteAccountAction, DeleteKeyAction,
+    DeployContractAction, FunctionCallAction, StakeAction, TransferAction,
+};
+use near_primitives::views::FinalExecutionOutcomeView;
+
+use crate::network::{CallExecution, CallExecutionDetails, Network, ViewResultDetails};
+use crate::rpc::client::{
+    send_batch_tx_and_retry, Client, DEFAULT_CALL_DEPOSIT, DEFAULT_CALL_FN_GAS,
+};
+use crate::types::{AccessKey, AccountId, Balance, Gas, InMemorySigner, PublicKey, SecretKey};
+use crate::worker::Worker;
+use crate::Account;
+
+/// A set of arguments we can provide to a transaction, containing
+/// the function name, arguments, the amount of gas to use and deposit.
+#[derive(Debug, Clone)]
+pub struct Function {
+    pub function: String,
+    pub args: Vec<u8>,
+    pub deposit: Balance,
+    pub gas: Gas,
+}
+
+impl Function {
+    pub fn new(function: &str) -> Self {
+        Self {
+            function: function.into(),
+            args: serde_json::json!({}).to_string().into_bytes(),
+            deposit: DEFAULT_CALL_DEPOSIT,
+            gas: DEFAULT_CALL_FN_GAS,
+        }
+    }
+
+    pub fn args(mut self, args: Vec<u8>) -> Self {
+        self.args = args;
+        self
+    }
+
+    pub fn args_json<U: serde::Serialize>(mut self, args: U) -> anyhow::Result<Self> {
+        self.args = serde_json::to_vec(&args)?;
+        Ok(self)
+    }
+
+    pub fn args_borsh<U: borsh::BorshSerialize>(mut self, args: U) -> anyhow::Result<Self> {
+        self.args = args.try_to_vec()?;
+        Ok(self)
+    }
+
+    pub fn deposit(mut self, deposit: u128) -> Self {
+        self.deposit = deposit;
+        self
+    }
+
+    pub fn gas(mut self, gas: u64) -> Self {
+        self.gas = gas;
+        self
+    }
+}
+
+impl From<Function> for Action {
+    fn from(args: Function) -> Self {
+        Self::FunctionCall(FunctionCallAction {
+            method_name: args.function,
+            args: args.args,
+            deposit: args.deposit,
+            gas: args.gas,
+        })
+    }
+}
+
+pub struct Transaction<'a> {
```

Comment:
wdyt about naming this something like `BatchTransaction`. Esp since we have other types of `Transaction`-s: `CallTransaction`, `CreateAccountTransaction`.

Reply:
I didn't go with that because Transactions are BatchTransactions. Like there's just a bunch of actions in a single transaction, plus other places such as near-api-js and nearcore call it purely just `Transaction`, so thought it be best to stay consistent that way

Reply:
Agreed. It's just because we have different types of `Transactions`, I thought it'll make sense to distinguish. I will mark this as resolved.